x/poolmanager/routes_test.go

```diff
@@ -0,0 +1,446 @@
+		t.Errorf("Unexpected route: %+v", routes[0][0])
+	}
+
+	if routes[0][1].PoolId != 3 || routes[0][1].Token != "token3" {
+		t.Errorf("Unexpected route: %+v", routes[0][1])
+	}
+
+	if routes[0][2].PoolId != 5 || routes[0][2].Token != "token4" {
+		t.Errorf("Unexpected route: %+v", routes[0][2])
+	}
+}
+
+func (s *KeeperTestSuite) TestGetSetDenomPairRoutes() {
+	// Set routes in state
+	routingGraph, err := s.App.PoolManagerKeeper.SetDenomPairRoutes(s.Ctx)
+	s.Require().NoError(err)
+	s.Require().Empty(routingGraph)
+
+	// Get routes from state and compare to expected
+	routingMap, err := s.App.PoolManagerKeeper.GetRouteMap(s.Ctx)
+	s.Require().NoError(err)
+	expectedRoutingMap := poolmanager.ConvertToMap(&routingGraph)
+	s.Require().Equal(expectedRoutingMap, routingMap)
+
+	// Prepare pools
+	s.PrepareBalancerPoolWithCoins(sdk.NewCoin("uosmo", sdk.NewInt(10000000000)), sdk.NewCoin("bar", sdk.NewInt(10000000000)))
+	s.PrepareConcentratedPoolWithCoins("uosmo", "foo")
+	s.PrepareCustomTransmuterPool(s.TestAccs[0], []string{"uosmo", "uion"})
+
+	// Create a pool that determines what the value of uosmo is in terms of USDC
+	poolManagerParams := s.App.PoolManagerKeeper.GetParams(s.Ctx)
+	poolManagerParams.MinValueForRoute.Denom = "usdc"
+	s.App.PoolManagerKeeper.SetParams(s.Ctx, poolManagerParams)
+	osmoUsdPool := s.PrepareConcentratedPoolWithCoins("uosmo", s.App.PoolManagerKeeper.GetParams(s.Ctx).MinValueForRoute.Denom)
+	s.CreateFullRangePosition(osmoUsdPool, sdk.NewCoins(sdk.NewCoin("uosmo", sdk.NewInt(10000000000)), sdk.NewCoin("usdc", sdk.NewInt(10000000000))))
+
+	// Set routes in state
+	routingGraph, err = s.App.PoolManagerKeeper.SetDenomPairRoutes(s.Ctx)
+	s.Require().NoError(err)
+	s.Require().NotEmpty(routingGraph)
+
+	// Get routes from state and compare to expected
+	routingMap, err = s.App.PoolManagerKeeper.GetRouteMap(s.Ctx)
+	s.Require().NoError(err)
+	expectedRoutingMap = poolmanager.ConvertToMap(&routingGraph)
+	s.Require().Equal(expectedRoutingMap, routingMap)
+
+	// 3 entries are expected, since foo and uion pools do not reach the min liquidity threshold
+	s.Require().Equal(3, len(routingGraph.Entries))
+
+	// Change min liquidity threshold to 0
+	poolManagerParams = s.App.PoolManagerKeeper.GetParams(s.Ctx)
+	poolManagerParams.MinValueForRoute.Amount = sdk.NewInt(0)
+	s.App.PoolManagerKeeper.SetParams(s.Ctx, poolManagerParams)
+
+	// Set routes in state
+	routingGraph, err = s.App.PoolManagerKeeper.SetDenomPairRoutes(s.Ctx)
+	s.Require().NoError(err)
+	s.Require().NotEmpty(routingGraph)
+
+	// Get routes from state and compare to expected
+	routingMap, err = s.App.PoolManagerKeeper.GetRouteMap(s.Ctx)
+	s.Require().NoError(err)
+	expectedRoutingMap = poolmanager.ConvertToMap(&routingGraph)
+	s.Require().Equal(expectedRoutingMap, routingMap)
+
+	// 5 entries for all denoms (uosmo, uion, bar, foo, usdc)
+	s.Require().Equal(5, len(routingGraph.Entries))
+}
+
+func (s *KeeperTestSuite) TestGetDirectRouteWithMostLiquidity() {
```

Comment:
Can we also add test case for this method where getDirectRouteWithMostLiquidity does not return any route?